Refactoring of Deadline to ReminderDate

kerbysoh opened the pull request:
- Refactor class name of Deadline to ReminderDate to more clearly highlight the purpose of the field within the role.

- ReminderDate is now an optional field within the addRoleCommand. When users opt not to key in a reminder date, the ReminderDate field within the Role would be created with its value field set to null.

- A ReminderDate with a null value field is a valid ReminderDate.

- Updated storage to parse null-valued ReminderDate as empty strings

- Update UI to display "N/A" when any role field is empty.

- Update UG, DG and relevant UML diagrams with the newly refactored names

kerbysoh committed:
Refactoring of Deadline to ReminderDate

- Refactor class name of Deadline to ReminderDate to more clearly highlight the purpose of the field within the role.

- ReminderDate is now an optional field within the addRoleCommand. When users opt not to key in a reminder date, the ReminderDate field within the Role would be created with its value field set to null.

- A ReminderDate with a null value field is a valid ReminderDate.

- Updated storage to parse null-valued ReminderDate as empty strings

- Update UI to display "N/A" when any role field is empty.

- Update UG, DG and relevant UML diagrams with the newly refactored names

kerbysoh commented:
Possible grammatical and wording errors where deadline is more appropriate than reminder date.

petermonky commented on docs/UserGuide.md:
```diff
@@ -245,8 +247,6 @@ Colour coding of roles allows users to quickly obtain information about their ap
 * "complete": black
 
-Format: `exit`
-
```
Nice fix here 👍

petermonky commented on src/main/java/seedu/tinner/logic/parser/CliSyntax.java:
```diff
@@ -13,7 +13,7 @@
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_STATUS = new Prefix("s/");
-    public static final Prefix PREFIX_DEADLINE = new Prefix("b/");
+    public static final Prefix PREFIX_REMINDER_DATE = new Prefix("b/");
```
Perhaps we can change the prefix to `r/`? And somehow unify both company name and role name prefixes as `n/`.

justinyjt replied:
>Perhaps we can change the prefix to r/

I agree with Peter on this, makes more intuitive sense for the user if reminder date had the r/ prefix.

>unify both company name and role name prefixes as n/

I believe that these prefixes are for the find command where you need to specify both company and role name in the same command, hence the distinction between the two.

kerbysoh replied:
I agree. Will make relevant changes.

petermonky replied:
> I believe that these prefixes are for the find command where you need to specify both company and role name in the same command, hence the distinction between the two.


Ahhh yep forgot about the find command, using `n/` for both could definitely pose an issue then. Perhaps could use some other prefixes then.

kerbysoh replied:
Yes I have retained the use of prefixes for the find command using c/ and r/. Only the addRole and editRole command has been edited, where the reminderDate field is indicated with a prefix r/.

petermonky reviewed (approved):
Clean refactoring 👍 LGTM

justinyjt commented on src/main/java/seedu/tinner/model/role/ReminderDate.java:
```diff
@@ -53,35 +60,46 @@ public static boolean isValidDeadline(String test) {
+        if (this.value == null) {
+            return false;
+        }
         LocalDateTime oneWeekAway = LocalDateTime.now().plusDays(reminderWindow);
```
Spotted a mini bug of my own here. I forgot to change the LocalDateTime variable name to oneReminderWindowAway. Could you help to just make a quick change to this for me so I don't need to make a whole PR just for that one variable 😅 . Thanks!

kerbysoh replied:
Sure no worries

justinyjt commented on src/main/java/seedu/tinner/storage/JsonAdaptedRole.java:
```diff
@@ -80,16 +84,14 @@ public Role toModelType() throws IllegalValueException {
         final Status roleStatus = new Status(status);
 
-        if (deadline == null) {
+        if (reminderDate == null) {
```
Based on our previous implementation, we removed null checks for optional fields as it is possible for these fields to be null. Should this apply to reminder date as well?

kerbysoh replied:
Can someone weigh in on this? @petermonky @BrendonLau @jinnhl? I believe that with our current implementation none of those optional fields will be null. They are empty strings but not null values. Correct me if I'm wrong.

petermonky replied:
> I believe that with our current implementation none of those optional fields will be null.

Does this apply to other classes as well? If so then I think we can add `null`checks the other classes too, just to be more defensive.

kerbysoh replied:
Yes I have updated null checks for them. Testing has yet to be updated.

justinyjt commented on src/test/java/seedu/tinner/storage/JsonAdaptedRoleTest.java:
```diff
@@ -56,47 +57,47 @@ public void toModelType_nullRoleName_throwsIllegalValueException() {
 
     @Test
-    public void toModelType_nullDeadline_throwsIllegalValueException() {
+    public void toModelType_nullReminderDate_throwsIllegalValueException() {
```
Refer to previous comment on JsonAdaptedRole: this test case may not be required.

justinyjt reviewed (commented):
Overall LGTM. Looks like you managed to capture most of the changes despite this being quite a large refactoring. Nice work :)

kerbysoh committed:
Refactor the use of "b/" prefix for the addRole and editRole commands to "r/"

- Prefix "r/" is used to represent reminder date.

- Make relevant changes in the DG, UG and UML Diagrams with respect to this change

- Enforce null checks for JsonAdaptedRole

- Minor changes to DG and UG to improve quality and fix grammatical mistakes
